src/main/java/org/terasology/lost/LevelSpawnSystem.java

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.lost;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.management.AssetManager;
+import org.terasology.biomesAPI.OnBiomeChangedEvent;
+import org.terasology.entitySystem.entity.EntityBuilder;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.inventory.InventoryManager;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.lost.generator.LostWorldGenerator;
+import org.terasology.math.Region3i;
+import org.terasology.math.Side;
+import org.terasology.math.geom.ImmutableVector2f;
+import org.terasology.math.geom.Vector2f;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.polyworld.graph.GraphFacet;
+import org.terasology.polyworld.graph.Region;
+import org.terasology.registry.In;
+import org.terasology.structureTemplates.events.SpawnStructureEvent;
+import org.terasology.structureTemplates.util.BlockRegionTransform;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.generation.facets.SurfaceHeightFacet;
+
+import java.util.Set;
+
+/**
+ * Creates the levels/challenges for the Lost gameplay. On every biome change event it is checked whether the it is
+ * suitable for a level spawn. If yes, the challenge corresponding to the biome is spawned in the centre of the vornoi
+ * {@link Region} entered The challenges start spawning in their respective biomes once the book in the well has been
+ * discovered
+ */
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class LevelSpawnSystem extends BaseComponentSystem {
+    @In
+    private EntityManager entityManager;
+    @In
+    private InventoryManager inventoryManager;
+    @In
+    private AssetManager assetManager;
+    @In
+    private WorldProvider worldProvider;
+    private static final Logger logger = LoggerFactory.getLogger(LevelSpawnSystem.class);
+    // to prevent overlapping with hut
+    int minimumDistanceFromHut = 30;
```

Comment:
While this one should probably be a `private static final int` in all caps and grouped with its new buddy the logger